### PR TITLE
feat: Allow lens runtime selection via config

### LINKED ISSUE #2683: Allow configuration of Lens runtime type
Like we do for ACP and rootstore

### PR DESCRIPTION
**Pull request by AndrewSisley**
## Relevant issue(s)

Resolves #2683

## Description

Allows lens runtime selection via config/cli param.

Also adds CI jobs to the matrix to test wasmer and wazero.  I was hoping it would solve the windows build issue, but still no joy there :(

### CODE REVIEW

File: node/lens.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+	"github.com/lens-vm/lens/host-go/engine/module"
+
+	"github.com/sourcenetwork/defradb/client"
+	"github.com/sourcenetwork/defradb/internal/lens"
+)
+
+type LensRuntimeType string
+
+const (
+	DefaultLens LensRuntimeType = ""
```
Comment (islamaliev) [inline node/lens.go:25]: todo: please give it a reasonable value of add a documentation why there is only 1 value here and why it has to be empty string.

File: node/lens.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+}
+
+// DefaultACPOptions returns new options with default values.
+func DefaultLensOptions() *LensOptions {
+	return &LensOptions{
+		lensPoolSize: lens.DefaultPoolSize,
+	}
+}
+
+type LenOpt func(*LensOptions)
```
Comment (islamaliev) [inline node/lens.go:45]: todo: it looks like `LenOpt` and `Option` are the same. Use one of them please

**Review by islamaliev — APPROVED** [review]
LGTM! I just have some minor change requests

File: .github/workflows/test-and-upload-coverage.yml [hunk 3]
```diff
@@ -63,6 +64,20 @@ jobs:
 ##          mutation-type: collection-save
 ##          detect-changes: false
 ##          database-encryption: false
+          - os: ubuntu-latest
+            client-type: go
+            database-type: badger-memory
+            mutation-type: collection-save
+            lens-type: wazero
```
[the remark this answers is not in the document]
Comment (AndrewSisley) [reply .github/workflows/test-and-upload-coverage.yml:71]: Yeah, I'm pretty sure I updated the description yesterday once I failed to get that working in the CI - doesn't seem to have saved though.

File: node/lens.go (hunk 1 above)
Comment (AndrewSisley) [reply node/lens.go:25]: Will do

- [ ] Doc default + map

File: node/lens.go (hunk 2 above)
Comment (islamaliev) [reply node/lens.go:45]: I see, thanks for the explanation

File: cli/start.go [hunk 4]
```diff
@@ -126,6 +126,7 @@ func MakeStartCommand() *cobra.Command {
 				http.WithAllowedOrigins(cfg.GetStringSlice("api.allowed-origins")...),
 				http.WithTLSCertPath(cfg.GetString("api.pubKeyPath")),
 				http.WithTLSKeyPath(cfg.GetString("api.privKeyPath")),
+				node.WithLensRuntime(node.LensRuntimeType(cfg.GetString("lens.runtime"))),
```
Comment (nasdf) [inline cli/start.go:129]: todo: can you add an entry to `docs/config.md` for this?

**Review by nasdf — CHANGES_REQUESTED** [review]
LGTM. Nice to see the node options expanded. One small documentation todo.

File: cli/start.go (hunk 4 above)
Comment (AndrewSisley) [reply cli/start.go:129]: Will do, thanks Keenan!

- [ ] Have a look at config.md and add an entry

**Review by nasdf — APPROVED** [review]

**Review by shahzadlone — APPROVED** [review]
LGTM

Adding these new required job names to merge the PR

**Pull request merged**